packages/commonwealth/server/util/verifySessionSignature.ts

```diff
@@ -91,8 +91,17 @@ const verifySessionSignature = async (
         });
         if (!user || !user.id) throw new Error('Failed to create user');
         addressModel.user_id = user!.id;
+
+        await addressModel.save();
+        return;
       }
     }
+
+    // user already exists but new community joined
+    await incrementProfileCount(
```

Comment:
Should we be calling the join community command from here? Seems like we have logical coupling here in the command and the verification module.

Reply:
Yeah I agree the flow is confusing. Basically this is the POST /verifyAddress which is called after the POST /createAddress. This flow happens when a user creates a new address.

the joinCommunity command replaced the POST /linkExistingAddressToCommunity which is used for linking existing addresses to a community. The problem is, this is a separate flow which is ran when the address joins any community after the first.

Reply:
Approving, but I think both entry points should call command that encapsulates all business logic, including updating the count projection